core/src/main/scala/bekindrewind/storage/FileVcrStorage.scala

```diff
@@ -0,0 +1,30 @@
+package bekindrewind.storage
+
+import bekindrewind.VcrEntries
+import bekindrewind.codec.{ CirceCodec, Codec }
+
+import java.nio.charset.{ Charset, StandardCharsets }
+import java.nio.file.{ Files, Path }
+import scala.util.Try
+
+class FileVcrStorage(path: Path, codec: Codec, charset: Charset) extends VcrStorage {
+  def read(): Either[Throwable, VcrEntries] =
+    for {
+      text    <- Try(new String(Files.readAllBytes(path), charset)).toEither
+      entries <- codec.decode(text)
+    } yield entries
+
+  def write(entries: VcrEntries): Unit = {
+    // Ensure directory exists
+    Option(path.getParent).foreach { directory =>
+      Files.createDirectories(directory)
+    }
+
+    Files.write(path, codec.encode(entries).getBytes(charset))
+  }
+}
+
+object FileVcrStorage {
+  def apply(recordingPath: Path): FileVcrStorage =
+    new FileVcrStorage(recordingPath, CirceCodec, StandardCharsets.UTF_8)
```

Comment:
To kepp YAML support in mind, maybe implicit parameter is better ?
```scala
  def apply(recordingPath: Path)(implicit codec: Codec): FileVcrStorage =
    new FileVcrStorage(recordingPath, codec, StandardCharsets.UTF_8)

```

Reply:
I'm not sure if it needs to be implicit. Maybe even default parameters (default of json) could work? I know some people don't like default parameters, but I'm already using them in the clients.

Also, maybe `CirceCodec` should be renamed to `JsonCodec`? Because circe also has a yaml library (https://github.com/circe/circe-yaml), which I imagine would be natural to use in this library. So we'd also have a `YamlCodec` in the future, right?

Reply:
Renamed to JsonCodec https://github.com/reibitto/be-kind-rewind/pull/23/commits/769861394878fac3e359ad66fcc396a8dda04988
And introduced default parameters https://github.com/reibitto/be-kind-rewind/pull/23/commits/48c9df02a2caf85c584ccfd3b2be99cfe8649297